language/move-stdlib/nursery/BitVector.move

```diff
@@ -0,0 +1,225 @@
+module 0x1::BitVector {
+    use 0x1::Vector;
+
+    /// The provided index is out of bounds
+    const EINDEX: u64 = 0;
+
+    struct BitVector has copy, drop, store {
+        length: u64,
+        bit_field: vector<u128>,
```

Comment:
Wondering: what motivated the choice to use u128? My (perhaps naive) thinking is that: `u8` or `u64` might be better for both on-disk/in-memory space efficiency and fewer instructions to execute on 64 bit machines.

Reply:
No particular reason other than trying to minimize the number of chunks, but I think moving to a `u8` might be a good idea here.

Reply:
Ah, chunk size is a fair concern that I did not consider; each chunk costs a couple of extra Move instructions to bump the loop counter and read another array index. That might very well outweigh the execution time savings of having a chunk fit in a register.

If it's not too much work, it might be interesting to benchmark a few variants against vectors of a few different sizes and see if anything jumps out from either an execution time or a gas cost perspective. I'm guessing that both the execution and storage differences between the different approaches will be negligible, but you never know...

Reply:
I'm guessing that u64 is the sweet spot. It will have 8x fewer chunks than u8 and also fit in a register

Reply:
I think you're intuition is correct Bob -- I timed each test, and u8 and u64 were pretty similar to each other in terms of performance (expected with the size of things), however,  the u64 implementation was slightly faster across the board, so I've updated the underlying representation to use u64s.